scwconfig/config.go

```diff
@@ -0,0 +1,323 @@
+
+	withProfile string
+}
+
+type profile struct {
+	AccessKey             *string `yaml:"access_key"`
+	SecretKey             *string `yaml:"secret_key"`
+	ApiUrl                *string `yaml:"api_url"`
+	Insecure              *bool   `yaml:"insecure"`
+	DefaultOrganizationId *string `yaml:"default_region"`
+	DefaultRegion         *string `yaml:"default_zone"`
+	DefaultZone           *string `yaml:"default_organization_id"`
+}
+
+func unmarshalConfV2(content []byte) (*configV2, error) {
+	var config configV2
+
+	err := yaml.Unmarshal(content, &config)
+	if err != nil {
+		return nil, err
+	}
+	return &config, nil
+
+}
+
+func (c *configV2) catchInvalidProfile() (*configV2, error) {
+	activeProfile, err := c.getActiveProfile()
+	if err != nil {
+		return nil, err
+	}
+
+	_, exist := c.Profiles[activeProfile]
+	if !exist {
+		return nil, fmt.Errorf("profile %s does not exist %s", activeProfile, inConfigFile())
+	}
+	return c, nil
+}
+
+func (c *configV2) getActiveProfile() (string, error) {
+	switch {
+	case c.withProfile != "":
+		return c.withProfile, nil
+	case os.Getenv(scwActiveProfileEnv) != "":
+		return os.Getenv(scwActiveProfileEnv), nil
+	case c.ActiveProfile != nil:
+		if *c.ActiveProfile == "" {
+			return "", fmt.Errorf("empty active profile %s", inConfigFile())
+		}
+		return *c.ActiveProfile, nil
+	default:
+		return "", nil
+	}
+}
+func (c *configV2) GetAccessKey() string {
+	// STEP 1: getenv
+	accessKey := getenv(scwAccessKeyEnv, terraformAccessKeyEnv)
+	if accessKey != "" {
+		return accessKey
+	}
+
+	// STEP 2: active profile
+	activeProfile, _ := c.getActiveProfile()
+	if activeProfile != "" && c.Profiles[activeProfile].AccessKey != nil {
+		return *c.Profiles[activeProfile].AccessKey
+	}
+
+	// STEP 3: default profile
+	if c.AccessKey != nil {
+		return *c.AccessKey
+	}
+	return ""
+}
+
+func (c *configV2) GetSecretKey() string {
+	// STEP 1: getenv
+	secretKey := getenv(scwSecretKeyEnv, cliSecretKeyEnv, terraformSecretKeyEnv, terraformAccessKeyEnv)
+	if secretKey != "" {
+		return secretKey
+	}
+
+	// STEP 2: active profile
+	activeProfile, _ := c.getActiveProfile()
+	if activeProfile != "" && c.Profiles[activeProfile].SecretKey != nil {
+		return *c.Profiles[activeProfile].SecretKey
+	}
+
+	// STEP 3: default profile
+	if c.SecretKey != nil {
+		return *c.SecretKey
+	}
+	return ""
+
+}
+
+func (c *configV2) GetApiUrl() string {
+	// STEP 1: getenv
+	apiUrl := getenv(scwApiUrlEnv)
+	if apiUrl != "" {
+		return apiUrl
+	}
+
+	// STEP 2: active profile
+	activeProfile, _ := c.getActiveProfile()
+	if activeProfile != "" && c.Profiles[activeProfile].ApiUrl != nil {
+		return *c.Profiles[activeProfile].ApiUrl
+	}
+
+	// STEP 3: default profile
+	if c.ApiUrl != nil {
+		return *c.ApiUrl
+	}
+	return ""
+}
+
+func (c *configV2) GetInsecure() bool {
+	// STEP 1: getenv
+	insecure := getenv(scwInsecureEnv, cliTLSVerifyEnv)
+	if insecure != "" {
+		insecureBool, _ := strconv.ParseBool(insecure)
```

Comment:
Why don't you return an error if `insecure` isn't a valid bool?
This would make things more verbose for the user.

Reply:
Those getters should not return error. Good point though, we will add a line of log here when we add the logger to the SDK.